fabtests/unit/mr_cache_evict.c

```diff
@@ -345,14 +431,24 @@ static void *mem_alloc(enum alloc_type type)
 /* MR registration function which returns the MR and the elapsed time, in
  * nanoseconds, to register the MR.
  */
-static int mr_register(const void *buf, struct fid_mr **mr, int64_t *elapsed)
+static int mr_register(const void *buf, struct fid_mr **mr, int64_t *elapsed,
+		       enum alloc_type type)
 {
-	uint64_t access = ft_info_to_mr_access(fi);
 	int ret;
+	const struct iovec iov = {
+		.iov_base = (void *) buf,
+		.iov_len = mr_buf_size,
+	};
+	struct fi_mr_attr mr_attr = {
+		.mr_iov = &iov,
+		.iov_count = 1,
+		.access = ft_info_to_mr_access(fi),
+		.requested_key = FT_MR_KEY,
+		.iface = type == CUDA ? FI_HMEM_CUDA : FI_HMEM_SYSTEM,
```

Comment:
Please put this conversion into a function, to make it easier to extend later.  e.g.
```
enum fi_hmem_iface get_iface(enum alloc_type)
{
    switch (type) {
    case CUDA:
        return FI_HMEM_CUDA;
    default:
        return FI_HMEM_SYSTEM;
    }
}
```